integrations/activewindow.cpp

```diff
@@ -0,0 +1,237 @@
+    explicit ActiveWindowWatcher(QObject *parent = nullptr);
+
+public slots:
+    Q_SCRIPTABLE void UpdateTitle(const QString &title);
+    Q_SCRIPTABLE void UpdateAttributes(const QString &json);
+
+
+private:
+    bool registerKWinScript();
+
+    Sensor *m_sensor;
+    QDBusInterface *m_kwinIface = nullptr;
+    QString m_lastTitle;
+    QString m_scriptPath;
+    bool m_connected = false;
+};
+
+
+ActiveWindowWatcher::ActiveWindowWatcher(QObject *parent)
+    : QObject(parent)
+{
+    m_sensor = new Sensor(this);
+    m_sensor->setId("active_window");
+    m_sensor->setName("Active Window");
+    m_sensor->setDiscoveryConfig("icon", "mdi:application");
+    // Register DBus service first
+    if (!QDBusConnection::sessionBus().registerService("org.davidedmundson.kiot.ActiveWindow")) {
+        qWarning() << "ActiveWindowWatcher: Failed to register DBus service";
+        m_sensor->setState("Unavailable");
+        return;
+    }
+
+    if (!QDBusConnection::sessionBus().registerObject("/ActiveWindow",
+            "org.davidedmundson.kiot.ActiveWindow",
+            this,
+            QDBusConnection::ExportAllSlots)) {
+        qWarning() << "ActiveWindowWatcher: Failed to register DBus object";
+        m_sensor->setState("Unavailable");
+        return;
+    }
+
+    if (!registerKWinScript()) {
+        qWarning() << "ActiveWindowWatcher: Failed to register KWin script";
+        m_sensor->setState("Unavailable");
+        return;
+    }
+}
+
+bool ActiveWindowWatcher::registerKWinScript()
+{
+    m_kwinIface = new QDBusInterface(
+        "org.kde.KWin",
+        "/Scripting",
+        "org.kde.kwin.Scripting",
+        QDBusConnection::sessionBus(),
+        this
+    );
+
+    if (!m_kwinIface->isValid()) {
+        qWarning() << "ActiveWindowWatcher: KWin scripting interface not available";
+        return false;
+    }
+
+    // Clean up any existing instance
+    QDBusMessage isLoadedReply = m_kwinIface->call("isScriptLoaded", "kiot_activewindow");
+    if (isLoadedReply.type() != QDBusMessage::ErrorMessage && !isLoadedReply.arguments().isEmpty()) {
+        bool isLoaded = isLoadedReply.arguments().first().toBool();
+        if (isLoaded)
+            m_kwinIface->call("unloadScript", "kiot_activewindow");
+    }
+
+    QString runtimeDir = QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation);
+    if (runtimeDir.isEmpty())
+        runtimeDir = "/tmp";
+    QString scriptDir = runtimeDir + "/kiot";
+    QDir().mkpath(scriptDir);
+    m_scriptPath = scriptDir + "/kwin_activewindow.js";
+
+    QFile scriptFile(m_scriptPath);
+    if (!scriptFile.open(QIODevice::WriteOnly | QIODevice::Truncate)) {
+        qWarning() << "ActiveWindowWatcher: could not open script file" << m_scriptPath;
+        return false;
+    }
+
+    QTextStream out(&scriptFile);
+    out << "var lastPayload = {};\n"
+        "var currentWindow = null;\n"
+        "\n"
+        "function updateActiveWindow(w) {\n"
+        "    if (!w) return;\n"
```

Comment:
something for the future.

You probably want some code to get the transient parent of w. 

Otherwise whenever we right click on something, we'll get the details of the popup not the app we care about.

If the goal was to make the lights go green when you were using Kate, that would be annoying.